### DISCUSSION TITLE
Inconsistent decompilation

### DISCUSSION THREAD

**Discussion by RyanHope** [NationalSecurityAgency/ghidra#5833] [Q&A]
Quite often my disassembly is inconsistent. As you can see in the example below, both lines that start with "uVar2 =" should be identical in the disassembly.
![image](https://github.com/NationalSecurityAgency/ghidra/assets/141005/2a930df3-5baa-4f8f-867d-88bcfb9a58e1)

However in one location I get USHORT_00051f70 but in the other I *(ushort *)(iVar4 + 0x1f70).

Why is this happening? How do I fix it?
Disassembly below.

![image](https://github.com/NationalSecurityAgency/ghidra/assets/141005/f9a1cce2-637d-4e23-b174-e7de985c64c3)

**Comment by Tim---** [comment]
Hi,

Your problem seems related to call conventions.

Your assembly kind of looks like that:
```
r29 = 0x50000
r15 = [r29 + 0x1f70]
// some calls
r14 = [r29 + 0x1f70]
// some other calls
```

Ghidra can tell on the first occurence that `r15 = [0x51f70]` by doing the computation.
However, you have some function calls before r29 gets reused. 
Depending on the call conventions used, Ghidra cannot be sure that r29 is preserved by those calls.

What's weird is that there is apparently only one call convention ([__stdcall](https://github.com/NationalSecurityAgency/ghidra/blob/54e0ab1a3897dad020275d6395acfe0c56c3994d/Ghidra/Processors/V850/data/languages/V850.cspec)) for the V850 processor.
This convention specifies that r29 is "unaffected".

Can you check that the functions called inbetween (FUN_0012d6be and FUN_0012d708) are using the __stdcall convention ? Also make sure to "Commit Params/Return" on each and see if it helps.
